### PR TITLE
Work towards Control Transfers

### PR DESCRIPTION
**Pull request by networkfusion**
superceeds #142

### CODE REVIEW

File: src/Device.Net.UnitTests/IntegrationTestControlTransfer.cs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+using System.Threading.Tasks;
+using Usb.Net;
+using Usb.Net.Windows;
+namespace Device.Net.UnitTests
+{
+    [TestClass]
+    public class IntegrationTestControlTransfer
+    {
+        [TestMethod]
+        public async Task Get_status_from_DFU_device_using_contol_transfer()
```
Comment (MelbourneDeveloper) [inline src/Device.Net.UnitTests/IntegrationTestControlTransfer.cs:16]: @networkfusion beautiful!

This is what I need.

File: src/Device.Net.UnitTests/IntegrationTestControlTransfer.cs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            //const int STATE_DFU_MANIFEST = 0x07;
+            //const int STATE_DFU_MANIFEST_WAIT_RESET = 0x08;
+            //const int STATE_DFU_UPLOAD_IDLE = 0x09;
+            //const int STATE_DFU_ERROR = 0x0A;
+            //const int STATE_DFU_UPLOAD_SYNC = 0x91;
+            //const int STATE_DFU_UPLOAD_BUSY = 0x92;
+
+
+            //Specify the response part of the Message Contract
+            var expectedResult = new byte[] { 0x02 }; //lets try STATE_DFU_IDLE first (since we are in DFU mode)...
```
Comment (MelbourneDeveloper) [inline src/Device.Net.UnitTests/IntegrationTestControlTransfer.cs:111]: @networkfusion Perfect.

**Comment by fdidron** [comment]
Hi @networkfusion and @MelbourneDeveloper 

I would love to use this library but I need Control transfers for a new project I'm working on. Is there anything that needs to happen to merge this ? Happy to help in any way to make it happen

**Comment by MelbourneDeveloper** [comment]
@fdidron everyone is looking for control transfer and it's basically the #1 priority right now. It's just a question of coming up with the best design that will work across all platforms. If you have design, please log your suggestions here:

https://github.com/MelbourneDeveloper/Device.Net/issues/126

There is progress going on and I expect to have something working in the next couple of weeks.

**Comment by MelbourneDeveloper** [comment]
@fdidron also, please join in the conversation about this on [Discord](https://discord.gg/ZcvXARm)

File: src/Usb.Net/UsbControlRecipient.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
+﻿namespace Usb.Net
+{
+    /// <summary>
+    /// Defines constants that indicate the recipient of a USB control transfer. 
+    /// The recipient is defined in the setup packet of the control request. See Table 9.2 of section 9.3 of the Universal Serial Bus (USB) specification (<see cref="www.usb.org"/>).
+    /// </summary>
+    public enum UsbControlRecipient
```
Comment (MelbourneDeveloper) [inline src/Usb.Net/UsbControlRecipient.cs:7]: @networkfusion question...

Where do these enums and descriptions come from? Is this from some official documentation?

**Pull request merged**